packages/flutter/lib/src/material/predictive_back_page_transitions_builder.dart

```diff
@@ -159,130 +265,363 @@ class _PredictiveBackGestureDetectorState extends State<_PredictiveBackGestureDe
+  // a value that looks reasonable on most devices.
+  // See https://github.com/flutter/flutter/issues/97349.
+  static const double _kDeviceBorderRadius = 32.0;
+
+  // Eyeballed on a Pixel 9 running Android 16.
+  static const int _kCommitMilliseconds = 100;
+
+  // Since we don't know the device border radius, this provides a smooth
+  // transition between the default radius and the actual radius.
+  final Tween<double> _borderRadiusTween = Tween<double>(begin: _kDeviceBorderRadius, end: 0.0);
+
+  final Tween<double> _gapTween = Tween<double>(begin: _kMargin, end: 0.0);
+  final Tween<double> _scaleTween = Tween<double>(begin: _kMinScale, end: 1.0);
+  final Tween<double> _opacityTween = Tween<double>(begin: 1.0, end: 0.0);
+
+  late final AnimationController _commitController;
+  late final CurvedAnimation _commitAnimation;
+  late final Listenable _mergedAnimations;
+  late final Animation<double> _opacityAnimation;
+  late final Animation<double> _scaleAnimation;
+  late Animation<double> _scaleCommitAnimation;
+  late Animation<double> _xAnimation;
+  late Animation<Offset> _positionCommitAnimation;
+
+  double _lastXDrag = 0.0;
+  double _lastYDrag = 0.0;
+  double _lastScale = 1.0;
+
+  // This isn't done as an animation because it's based on the vertical drag
+  // amount, not the progression of the back gesture like widget.animation is.
+  double _getYPosition(double screenHeight) {
+    final double startTouchY = widget.startBackEvent?.touchOffset?.dy ?? 0;
+    final double currentTouchY = widget.currentBackEvent?.touchOffset?.dy ?? 0;
+
+    final double yShiftMax = (screenHeight / _kDivisionFactor) - _kMargin;
+
+    final double rawYShift = currentTouchY - startTouchY;
+    final double easedYShift =
+        Curves.easeOut.transform(clampDouble(rawYShift.abs() / screenHeight, 0.0, 1.0)) *
+        rawYShift.sign *
+        yShiftMax;
+
+    return clampDouble(easedYShift, -yShiftMax, yShiftMax);
+  }
+
+  Animation<Offset> _getCommitPositionAnimation(double screenWidth) {
+    return Tween<Offset>(
+      begin: Offset(_lastXDrag, _lastYDrag),
+      end: Offset(screenWidth * _kYPositionFactor, 0.0),
+    ).animate(_commitAnimation);
+  }
+
+  // TODO(justinmc): Should have a delegatedTransition to animate the incoming
+  // route regardless of its page transition.
+  // https://github.com/flutter/flutter/issues/153577
+
+  @override
+  void initState() {
+    super.initState();
+
+    _commitController = AnimationController(
+      duration: const Duration(milliseconds: _kCommitMilliseconds),
+      vsync: this,
+    );
+    _commitAnimation = CurvedAnimation(parent: _commitController, curve: Curves.easeOut);
+    _mergedAnimations = Listenable.merge(<Listenable>[widget.animation, _commitAnimation]);
+    _opacityAnimation = _opacityTween.animate(_commitAnimation);
+    _scaleAnimation = _scaleTween.animate(widget.animation);
+    _scaleCommitAnimation = Tween<double>(begin: _lastScale, end: 1.0).animate(_commitAnimation);
```

Comment:
we only really need this when phase is commit right? otherwise, we would just throw it away later.